skills/timezone.js

```diff
@@ -0,0 +1,65 @@
+/*
+
+ This simply hears the time in United States timezones and reports the system time (EST) and adjusts it accordingly
+
+*/
+
+module.exports = function(controller) {
+    //listening for time
+    controller.hears(
+        ['^time (.*)'],
+        'direct_message,direct_mention',
+        (bot, message) => {
+            const timezone = message.match[1];
+            const time = getTimeForTimezone(timezone);
+          bot.createConversation(message, (err, convo) => {
+            if (err) return false;
+            const responseMessage = 'The current time is ' + time + '.';
+            convo.say(responseMessage);
+            convo.activate();
+          });
+        });
+
+    //get system time
+    const getTimeForTimezone = function (timezone){
+        var date = new Date();
+        let hours = date.getHours();
+        let adjustedHours = hours + adjustForTimezone(timezone);
+        let minutes = (date.getMinutes()<10?'0':'') + date.getMinutes();
+        return adjustedHours + ':' + minutes;
+    }
+
+    //time zone adjustment
+    const adjustForTimezone = function (timezone){
```

Comment:
Any ideas on generalising this so it applies on other non-US timezones?

Reply:
Either hardcoding all of the timezones (expanding on what i currently have) or hitting some sort of api. I can't think of any in-between solution.